processing/src/main/java/org/apache/druid/segment/incremental/OnheapIncrementalIndex.java

```diff
@@ -431,6 +538,43 @@ private void closeAggregators()
     }
   }
 
+  @Nullable
+  @Override
+  public ProjectionRowSelector getProjection(CursorBuildSpec buildSpec)
+  {
+    if (buildSpec.getQueryContext().getBoolean(QueryContexts.CTX_NO_PROJECTION, false)) {
+      return null;
+    }
```

Comment:
I think that it is likely the only ever direct caller of this method is going to be `IncrementalIndexCursorFactory.makeCursorHolder`, which seems also a reasonable place to check, I can move it there if you prefer. The main reason I did it here is because i wired up both context flags I added at the same time when I wrote it, since here is also where checking the context flag for forcing a specific projection be used by name is happening.